internal/observability/events/event_audit.go

```diff
@@ -0,0 +1,142 @@
+package event
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/eventlogger"
+)
+
+const AuditVersion = "v0.1"
+
+type AuditEventType string
+
+const (
+	ApiRequest AuditEventType = "APIRequest"
+)
+
+type Audit struct {
+	Id             string       `json:"id"`                     // std audit/boundary field
+	Version        string       `json:"version"`                // std audit/boundary field
+	Type           string       `json:"type"`                   // std audit field
+	Timestamp      time.Time    `json:"timestamp"`              // std audit field
+	RequestInfo    *RequestInfo `json:"request_info,omitempty"` // boundary field
+	Auth           *Auth        `json:"auth,omitempty"`         // std audit field
+	Request        *Request     `json:"request,omitempty"`      // std audit field
+	Response       *Response    `json:"response,omitempty"`     // std audit field
+	SerializedHMAC string       `json:"serialized_hmac"`        // boundary field
+	Flush          bool         `json:"-"`
+}
+
+func newAudit(fromOperation Op, opt ...Option) (*Audit, error) {
```

Comment:
The entry point for writing Audit events will be: event.WriteAudit(ctx context.Context, a *Audit, opt...Option)  The Audit struct has to be exported since it's referenced in the exported WriteAudit(...), but newAudit(...) should never be called by anyone outside the event package.